### PR TITLE
Add a new perf job to Jenkins

### PR DESCRIPTION
**Pull request by DrewScoggins**
I am adding in a new job that will run our performance tests in Jenkins.  Just starting with Windows for now, and will add Linux later.

### CODE REVIEW

**Comment by DrewScoggins** [comment]
test ci please

**Comment by DrewScoggins** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -285,6 +285,41 @@ def osShortName = ['Windows 10': 'win10',
 }
 
 // **************************
+// Define perf testing.  These tests should be run daily, and will run on Jenkins not in Helix
+// **************************
+[true, false].each { isPR ->
```
Comment (ianhays) [inline netci.groovy]: if it's only being run as a daily triggered build, do you still need the `isPR` switch?

File: netci.groovy [hunk 2]
```diff
@@ -285,6 +285,41 @@ def osShortName = ['Windows 10': 'win10',
...
+            def newJob = job(Utilities.getFullJobName(project, newJobName, isPR)) {
+                // On Windows we use the packer to put together everything. On *nix we use tar
+                steps {
+                    batchFile("call \"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat\" x86 && build.cmd /p:ConfigurationGroup=${configurationGroup} /p:OSGroup=${osGroup} /p:Performance=true")
+                    batchFile("C:\\Packer\\Packer.exe .\\bin\\build.pack .\\bin")
+                }
+            }
+
+            // Set the affinity.
+            Utilities.setMachineAffinity(newJob, os, 'latest-or-auto')
```
Comment (ianhays) [inline netci.groovy]: Running daily perf tests on **any** machine will give unreliable results, won't it? It seems like you'd want some fixed perf machines that have the same hardware and configuration and don't run any other tests. If our between-build cleanup is *really* thorough then this probably won't affect our results too much, but it's worth considering.

**Comment by ianhays** [comment]
what's the benefit to adding a Jenkins daily perf build? Why not just add a daily trigger to the Helix job?

File: netci.groovy (hunk 2 above)
Comment (DrewScoggins) [inline netci.groovy]: Oh yeah for sure.  I wanted to first get the tests running and producing results.  Once we have that we can actually put them on the perf machines.  I just didn't want to do all of it at once.

**Comment by DrewScoggins** [comment]
Adding @dsgouda.

I am not sure what you mean by add a daily trigger to the Helix job?  We already have a daily run of the performance tests in Helix.  Is there a way that we can trigger a job to run in Jenkins from there?  My main goal here is to get our performance tests running inside Jenkins, and explore the feasibility of that scenario.

File: netci.groovy (hunk 2 above)
Comment (ianhays) [inline netci.groovy]: Gotcha, sounds good.

File: netci.groovy (hunk 1 above)
Comment (DrewScoggins) [inline netci.groovy]: No, we do not.  Eventually, we want to run on PRs, but for now we will take this out.

**Comment by ianhays** [comment]
>My main goal here is to get our performance tests running inside Jenkins, and explore the feasibility of that scenario.

That answers my question :) I was curious where the value was in adding a daily Jenkins run instead of using the daily Helix run results.

**Comment by DrewScoggins** [comment]
@dotnet-bot test ci please

**Comment by DrewScoggins** [comment]
@dotnet-bot test ci please

**Comment by stephentoub** [comment]
@DrewScoggins, can you comment on the status of this PR? Thanks.

**Comment by DrewScoggins** [comment]
I am going to try and finish this up this week.

**Comment by stephentoub** [comment]
Thanks.

**Comment by karelz** [comment]
@DrewScoggins what is the status?

**Comment by stephentoub** [comment]
@dotnet-bot test ci please

**Comment by stephentoub** [comment]
cc: @mmitche

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test ci please

**Comment by stephentoub** [comment]
@dotnet-bot test this please
@dotnet-bot test ci please

**Comment by Petermarcu** [comment]
@DrewScoggins , Any update here?

**Comment by DrewScoggins** [comment]
Yeah, I was working on this and got pulled on to finishing the CoreCLR Jenkins onboarding.  If you guys want I can just close this PR, and reopen a new one when I get back to this.

**Comment by stephentoub** [comment]
Thanks, Drew. Sounds good.

**Comment by karelz** [comment]
Yes, please! It doesn't help to have PRs opened when they are not converging anytime soon. Thanks!

**Pull request closed without merging**